### ISSUE TITLE
better to introduce browserHistory instead of hashHistory? (lesson 2)

### ISSUE THREAD

**Issue by cch5ng** [reactjs/react-router-tutorial#79]
in lesson 2, the Router history is initially set as hashHistory. later, we are instructed to change this to browserHistory (lesson 10). I was using react-router v2.0.0 and it gave me errors about not having any default history when using hashHistory (let me know if you need more precise wording). it seems better to be introduced to browserHistory.

**Comment by yormi** [comment]
No problem here with `hashHistory`... well at first yes but it was because of a typo :P

**Comment by cch5ng** [comment]
I will try to be more specific about steps to reproduce the error I first saw. I think in lesson 2, I did not have an issue but some lessons later, I saw an error in the console which linked to a react router doc about custom histories.

as a learner, it did not make sense to start with hash histories, then change to custom histories, and then end up with browser history. it would have been more clear to start with browser history.

my current state is at the end of lesson 11 and when I tried to replace the browserHistory with hashHistory, I did not see the original error which caused me to write this issue. I will try to find out more details to reproduce the error and see if this is an issue that would affect anyone else.

**Comment by yormi** [comment]
`Warning: [react-router] Router no longer defaults the history prop to hash history. Please use the `hashHistory` singleton instead. http://tiny.cc/router-defaulthistory`

Is that the error you had ? That's the one I had when I was importing and sending `hasHistory` (notice the typo) to the `Router`.

**Comment by cch5ng** [comment]
I am unable to reproduce the error. Closing the issue.

**Comment by ryanflorence** [comment]
We start with `hashHistory` because no matter what kind of server setup you've got (or maybe you dont' even have a server!) it will work. Browser history requires some finesse on your server.

**Comment by cch5ng** [comment]
@ryanflorence thanks for your response. before I received it, I had tried in the lesson 2 context to replace hashHistory with browserHistory (without config changes) and indeed, it did not work. also in the lesson 10 context, I tried replacing browserHistory with hashHistory and this did work (without any router errors). for those reasons, I decided to close the issue.

this is off the specific issue topic, but in the lesson about server rendering/routing, it would be nice to have a couple brief statements about the typical scenarios when it would be preferred to use server routing vs client routing. I tried to google this but didn't see any sources that immediately looked like credible resources but I guess I can read more deeply about it.

thanks again and thanks for the tutorial in general.
